### PR TITLE
forward ios console.logs

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 
I mentioned this on slack. This is a super short term solution to the lack of console.logs in the log send. The console.log will also capture the actionLogger (since that calls console.log also) which will help with our debugging but also means the logs are like 3x the size they need to be.

Longer term I think we need a better holistic solution. Currently

- Console.log overwriting is very messy and flow hates it and its hard to understand what's really going on
- We handle the logs in 3 different ways between desktop/android/ios

Instead I think we should have some explicit logger that has the semantics we really want and just use that instead of console.log. Let's also unify how this all works so the flow is 100% the same across the platforms

### CODE REVIEW

File: shared/settings/feedback-container.js [hunk 1]
```diff
@@ -49,19 +49,31 @@ class FeedbackContainer extends Component<void, {}, State> {
       // We don't get the notification from the daemon so we have to do this ourselves
       const logs = []
       console.log('Starting log dump')
-      dumpLoggers((...args) => {
+
+      const logNames = ['actionLogger', 'storeLogger']
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: explicitly dump these logs and append the console log

File: shared/settings/feedback-container.js [hunk 2]
```diff
@@ -49,19 +49,31 @@ class FeedbackContainer extends Component<void, {}, State> {
...
+          logger && logger.dumpAll((...args) => {
+            logs.push(args)
+          })
         } catch (_) {}
       })
 
+      logs.push(['=============CONSOLE.LOG START============='])
+      const logger = getLogger('iosConsoleLog')
+      logger && logger.dumpAll((...args) => {
+        logs.push([args[1], ...args.slice(2)])
```
Comment (chrisnojima) [inline shared/settings/feedback-container.js]: skip the prefixes the period logger injects

File: shared/util/forward-logs.native.js [hunk 3]
```diff
@@ -20,17 +20,17 @@ function setupSource () {
 
   window.console.log = (...args) => {
     localLog(...args)
-    logger.info(args.join(', '))
+    logger.info(...args)
```
Comment (chrisnojima) [inline shared/util/forward-logs.native.js]: just pass this through raw, the join happens on the android side

File: shared/util/logger.android.js [hunk 4]
```diff
@@ -1,3 +1,10 @@
 // @flow
 import {NativeModules} from 'react-native'
-export default NativeModules.FileLogger
+
+const logger = {
+  info: (...args: Array<any>) => NativeModules.FileLogger.info(args.join(',')),
```
Comment (chrisnojima) [inline shared/util/logger.android.js]: replicate the old behavior

File: shared/util/logger.ios.js [hunk 5]
```diff
@@ -1,6 +1,12 @@
...
+import {setupLogger} from './periodic-logger'
+
+const MAX = 10000
+const iosConsoleLogger = setupLogger('iosConsoleLog', MAX, false, null, 0, false)
+
 export default {
-  info: () => {},
-  warn: () => {},
-  error: () => {},
+  info: (...args: Array<any>) => iosConsoleLogger.log(...args),
```
Comment (chrisnojima) [inline shared/util/logger.ios.js]: forward to our logger

File: shared/util/periodic-logger.js [hunk 6]
```diff
@@ -16,45 +16,48 @@ class PeriodicLogger {
...
     this._size = size
     this._name = name
     this._logIncoming = logIncoming
+    this._logToConsole = logToConsole
     this._logTransform = logTransform || ((args: Array<any>) => args)
     this._logAfterXActions = logAfterXActions
   }
 
   _write (args: Array<any>) {
+    const argsWithTimestamp = [Date.now(), ...args]
```
Comment (chrisnojima) [inline shared/util/periodic-logger.js]: always add timestamp

File: shared/settings/feedback-container.js (hunk 2 above)
Comment (chromakode) [inline shared/settings/feedback-container.js]: Could you please add this explanation as a comment?

**Pull request merged**